R/summary_functions.R

```diff
@@ -133,6 +133,63 @@ zero_coverage <- function(x, ...) {
   }
 }
 
+#' Checks whether a branch contains an expression
+#' s1: a srcref of a branch
+#' s2: a srcref of an expression
+srcref_contains <- function (br_srcref, exp_srcref) {
+  s1 <- as.integer(br_srcref)
+  s2 <- as.integer(exp_srcref)
+
+  br_ln1 <- s1[[1L]]
+  br_col1 <- s1[[5L]]
+  br_ln2 <- s1[[3L]]
+  br_col2 <- s1[[6L]]
+
+  exp_ln1 <- s2[[1L]]
+  exp_col1 <- s2[[5L]]
+  exp_ln2 <- s2[[3L]]
+  exp_col2 <- s2[[6L]]
+
+  (br_ln1 < exp_ln1 | (br_ln1 == exp_ln1 & br_col1 <= exp_col1)) &
+    (br_ln2 > exp_ln2 | (br_ln2 == exp_ln2 & br_col2 >= exp_col2))
+}
+
+branch_coverage <- function (x) {
+  br_c <- attr(x, "branches")
+  attr(br_c, "relative") <- attr(x, "relative")
+  attr(br_c, "package") <- attr(x, "package")
+  class(br_c) <- "coverage"
+
+  br_c
+}
+
+#' Tally branch coverage
+#'
+#' @inheritParams print.branch_coverage
+#' @return a `data.frame` of branch coverage 
+#' @exports
+tally_branch_coverage <- function (x) {
```

Comment:
Would it be possible to somehow abstract over the tally_coverage and tally_branch_coverage so there only one function or they both call some common function? Just to reduce the code duplication.

Reply:
I am not sure. There doesn't seem to be much duplication. It looks to me what tally_coverage does is mostly formatting of the result of as.data.frame(coverage). It calls as.data.frame in the very beginning and the rest is to deal with NA rows and blank lines in the data.frame.

Reply:
tally_branch_coverage calculates covered branches and calls as.data.frame on the result as the last step.

Reply:
You are right, this looks good.